File: testdata/jennies/builders/discriminator_without_option/GoBuilder/discriminator_without_option/noshowfieldoption_builder_gen.go
package discriminator_without_option

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[NoShowFieldOption] = (*NoShowFieldOptionBuilder)(nil)

type NoShowFieldOptionBuilder struct {
    internal *NoShowFieldOption
    errors cog.BuildErrors
}

func NewNoShowFieldOptionBuilder() *NoShowFieldOptionBuilder {
	resource := NewNoShowFieldOption()
	builder := &NoShowFieldOptionBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}

	return builder
}



func (builder *NoShowFieldOptionBuilder) Build() (NoShowFieldOption, error) {
	if err := builder.internal.Validate(); err != nil {
		return NoShowFieldOption{}, err
	}
	
	if len(builder.errors) > 0 {
	    return NoShowFieldOption{}, cog.MakeBuildErrors("discriminator_without_option.noShowFieldOption", builder.errors)
	}

	return *builder.internal, nil
}

func (builder *NoShowFieldOptionBuilder) Text(text string) *NoShowFieldOptionBuilder {
    builder.internal.Text = text

    return builder
}

File: testdata/jennies/builders/discriminator_without_option/TypescriptBuilder/src/discriminatorWithoutOption/noShowFieldOptionBuilder.gen.ts
import * as cog from '../cog';
import * as discriminatorWithoutOption from '../discriminatorWithoutOption';

export class NoShowFieldOptionBuilder implements cog.Builder<discriminatorWithoutOption.NoShowFieldOption> {
    protected readonly internal: discriminatorWithoutOption.NoShowFieldOption;

    constructor() {
        this.internal = discriminatorWithoutOption.defaultNoShowFieldOption();
    }

    /**
     * Builds the object.
     */
    build(): discriminatorWithoutOption.NoShowFieldOption {
        return this.internal;
    }

    text(text: string): this {
        this.internal.text = text;
        return this;
    }
}
